Libs/Particles/ParticleShapeStatistics.cpp

```diff
@@ -208,6 +208,154 @@ int ParticleShapeStatistics::ImportPoints(std::vector<Eigen::VectorXd> points, s
   return 0;
 }
 
+//---------------------------------------------------------------------------
+void ParticleShapeStatistics::SetNumberOfParticlesAr(std::vector<int> num_particles_ar)

```

Comment:
What does "Ar" stand for?

Reply:
It means 'Array'. This holds the number of particles for each domain which is needed to compute multi-level statistics. I just added this suffix so that it won't be confused with num_particles already been used.